packages/indexer/native/src/lib.rs

```diff
@@ -398,6 +325,146 @@ declare_types! {
+                args_len as u32
+            };
+            let mut start_key = vec![prefix as u8];
+            start_key.extend_from_slice(script.code_hash().as_slice());
+            start_key.extend_from_slice(script.hash_type().as_slice());
+            start_key.extend_from_slice(&args_len.to_le_bytes());
+            start_key.extend_from_slice(&script.args().raw_data());
+
+            let from_block = cx.argument::<JsValue>(4)?;
+            let to_block = cx.argument::<JsValue>(5)?;
+            let from_block_number_slice = if from_block.is_a::<JsNumber>() {
+                let from_block_number = from_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64;
+                from_block_number.to_be_bytes()
+            } else {
+                0_u64.to_be_bytes()
+            };
+            let to_block_number_slice = if to_block.is_a::<JsNumber>() {
+                // here set to_block_number as toBlock + 1, making the toBlock included in query range.
+                let to_block_number = to_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64 + 1;
+                to_block_number.to_be_bytes()
+            } else {
+                u64::MAX.to_be_bytes()
+            };
+
+            let skip = cx.argument::<JsValue>(6)?;
+            let skip_number = if skip.is_a::<JsNumber>() {
+                skip.downcast::<JsNumber>().or_throw(&mut cx)?.value() as usize
+            } else {
+                0_usize
+            };
+
+            let iter = store.iter(&start_key, IteratorDirection::Forward);
+            if iter.is_err() {
+                return cx.throw_error("Error creating iterator!");
+            }
+
+            let iter = iter.unwrap()
+            .take_while(move |(key, _)| {
+                let block_number_slice = key[key.len() - 16..key.len() - 8].try_into();
+                key.starts_with(&start_key) && to_block_number_slice > block_number_slice.unwrap()
+            })
+            .filter( move |(key, _)| {
+                let block_number_slice = key[key.len() - 16..key.len() - 8].try_into();
+                from_block_number_slice <= block_number_slice.unwrap()
+            })
+            .skip(skip_number);
+
+            Ok(LiveCellIterator(Box::new(iter)))
+        }
+
+        method collect(mut cx) {
+            let mut this = cx.this();
+            let live_cell_key_values = {
+                let guard = cx.lock();
+                let mut iterator = this.borrow_mut(&guard);
+                let mut key_values = vec![];
+                while let Some((key, value)) = iterator.0.next() {
+                    key_values.push((key,value));
+                }
+                key_values
+            };
+            let js_out_points = JsArray::new(&mut cx, live_cell_key_values.len() as u32);
+            for (i, (key, value)) in live_cell_key_values.iter().enumerate() {
```

Comment:
You can perform the early stop checking at here.

Reply:
I don't get it. The `search` phase  would finish after line 432 - 434, I don't see what we can do here. A partial performance improvement I can do is separating the `JsLiveCellIterator` construct into two cases:
1. with `prefix search on args`, use the current version of iterator;
2. without `prefix search on args`, use last version of iterator, which doesn't suffer from the performance issue.

Reply:
Wrong location, see comments above.